### ISSUE TITLE
Highlighting of format_args! does not work with tracing macros

### ISSUE THREAD

**Issue by ian-h-chamberlain** [rust-lang/rust-analyzer#12405]
Syntax highlighting in macros like `println!` and `format_args!` is awesome! Unfortunately, it doesn't seem to work with `tracing` macros as I would expect it to, since the expansion appears to use `format_args!` similarly to other cases. Example:

![Screen Shot 2022-05-28 at 9 00 54 AM](https://user-images.githubusercontent.com/11131775/170826757-694a8baa-209f-4a67-a3de-9dbd07c0f717.png)

So far, I have not been able to reproduce this using my own macro definition, and the upstream `tracing::event` has a fairly big + complicated definition, but one match arm does use `format_args!` and the expansion contains this:
```rust
Some(
    &(unsafe {
        std::fmt::Arguments::new_v1(
            &[],
            &[std::fmt::ArgumentV1::new(
                &("not highlighted"),
                std::fmt::Display::fmt,
            )],
        )
    }) as &Value,
),
```

Please let me know if there's anything I can do to help track down why this expansion doesn't match the criteria for highlighting. Thanks!

----

**rust-analyzer version**: rust-analyzer version: 84be2eaf9 2022-05-23 stable

**rustc version**: rustc 1.61.0 (fe5b13d68 2022-05-18)

**relevant settings**: None that I know of

**Comment by ian-h-chamberlain** [comment]
I know this issue hasn't seen much movement since I first filed it, but I had some renewed interest after seeing #15559 in the recent weekly changelog. Good news: I was able to find a minimum viable reproduction!

```rust
macro_rules! nothing {
    ($tt:tt) => {};
}

macro_rules! expand_multiple {
    ($fmt:literal) => {{
        format_args!($fmt);
        nothing!($fmt);
    }};
}

fn main() {
    let some_val = "okay";

    expand_multiple!("reproduces: {some_val}");
}
```

When `format_args!` is _last_ in the expansion, highlighting works as expected. When `nothing!` is last, highlighting is disabled. It seems that the highlight logic is based on whichever expansion is encountered most recently while traversing.

I'm not sure if this is really supported, since of course macros can do whatever they want with the literal, but I'd suggest that usually, if _any_ part of the expansion is `format_args!($lit)` it is probably safe to assume the macro is intended to be used this way. Maybe the logic could be updated to turn on highlighting in this scenario? It seems this might affect other highlight use cases though so might be a bigger change than just for `format_args!`.

**Comment by Veykril** [comment]
Yes, syntax highlighting currently only looks at the first (as in iteration order) mapping it encounters for performance reasons. That might change in the future as we are rewriting some of the mapping stuff soon.

**Comment by SE2Dev** [comment]
Just out of curiosity, which PR / commit fixed this issue? Was it [#16027](https://github.com/rust-lang/rust-analyzer/pull/16027)?

I did confirm that the current pre-release version of `rust-lang.rust-analyzer` (`v0.4.1760`) *does* seem to fix this issue.

**Comment by Veykril** [comment]
Yes, https://github.com/rust-lang/rust-analyzer/commit/9b7ec5e31be75ed49b2cc7f341fd93ca6999a303 specifically
